### PR TITLE
use stdlib puppet facts for configuration

### PR DESCRIPTION
**Pull request by vchepkov**
puppetserver doesn't honor puppet.conf vardir settings

This would solve #351

### CODE REVIEW

**Comment by rnelson0** [comment]
This broke all the tests. Can you review the failures to see what the fix is?

**Comment by vchepkov** [comment]
As far as I can tell, all failures have nothing to do with my change, seems Travis environment is broken to begin with.

**Comment by rnelson0** [comment]
The errors are definitely related to your changes, they have to do with r10k_basedir. It's expecting an actual path and getting something else. It may be a missing fact or it may need to change the data type. I suggest rebasing on master first, though, there has been a modulesync since this was submitted.

**Comment by vchepkov** [comment]
I will be first one to admin I am bad with tests. I presume the problem is, actual puppet is not running there, so facts are not being collected. Do I add them to default_module_facts.yml  ?

**Comment by vchepkov** [comment]
Well, that didn't help :(

**Comment by rnelson0** [comment]
I think you need to set the facts hash element directly, rspec-puppet doesn't normalize between global facts and the facts hash.

**Comment by vchepkov** [comment]
How do I do this?

**Comment by vchepkov** [comment]
I found something in spec/spec_helper.rb, I'll try to add it there

**Comment by vchepkov** [comment]
I found this PR https://github.com/voxpupuli/puppet-selinux/pull/217/files
And it is using exactly the same changes, but their test succeeds.
I am not sure what I am doing wrong here

**Comment by rnelson0** [comment]
@vchepkov Can you rebase on master? We updated with a modulesync recently, at the least that should make the test results easier to read.

**Comment by vchepkov** [comment]
Done

File: default_module_facts.yml [hunk 1]
```diff
@@ -1,2 +1,4 @@
 ---
 pe_version: '2016.4.1'
+puppet_environmentpath: '/etc/puppetlabs/code/environments'
+puppet_vardir: '/opt/puppetlabs/puppet/cache'
```
Comment (rnelson0) [inline default_module_facts.yml]: Shouldn't this file go in `spec`? That's where [selinux](https://github.com/oranenj/puppet-selinux/blob/36a6b4f1ed540595d67257a972032a605c6321fa/spec/default_module_facts.yml) puts it.

Comment (vchepkov) [inline default_module_facts.yml]: I honestly don't know, I added to an existing file, I assumed it's in the right place
let me move it

**Comment by vchepkov** [comment]
Much better now

**Comment by rnelson0** [comment]
You can blame me for putting the file in the wrong spot :) Thanks!

**Pull request merged**